src/components/ImageView/index.native.js

```diff
@@ -102,148 +115,123 @@ class ImageView extends PureComponent {
      *
      * @param {Object} nativeEvent
      */
-    configureImageZoom({nativeEvent}) {
-        let imageWidth = nativeEvent.width;
-        let imageHeight = nativeEvent.height;
-        const containerWidth = Math.round(this.props.windowWidth);
-        const containerHeight = Math.round(this.state.containerHeight ? this.state.containerHeight : this.props.windowHeight);
+    const configureImageZoom = ({nativeEvent}) => {
+        let imgWidth = nativeEvent.width;
+        let imgHeight = nativeEvent.height;
+        const roundedContainerWidth = Math.round(windowWidth);
+        const roundedContainerHeight = Math.round(containerHeight || windowHeight);
 
-        const aspectRatio = Math.min(containerHeight / imageHeight, containerWidth / imageWidth);
+        const aspectRatio = Math.min(roundedContainerHeight / imgHeight, roundedContainerWidth / imgWidth);
 
-        imageHeight *= aspectRatio;
-        imageWidth *= aspectRatio;
+        imgHeight *= aspectRatio;
+        imgWidth *= aspectRatio;
 
         // Resize the image to max dimensions possible on the Native platforms to prevent crashes on Android. To keep the same behavior, apply to IOS as well.
         const maxDimensionsScale = 11;
-        imageWidth = Math.min(imageWidth, containerWidth * maxDimensionsScale);
-        imageHeight = Math.min(imageHeight, containerHeight * maxDimensionsScale);
-        this.setState({imageHeight, imageWidth, isLoading: false});
-    }
+        imgWidth = Math.min(imgWidth, roundedContainerWidth * maxDimensionsScale);
+        imgHeight = Math.min(imgHeight, roundedContainerHeight * maxDimensionsScale);
 
-    /**
-     * When the url changes and the image must load again,
-     * this resets the zoom to ensure the next image loads with the correct dimensions.
-     */
-    resetImageZoom() {
-        if (this.imageZoomScale !== 1) {
-            this.imageZoomScale = 1;
+        setImageHeight(imgHeight);
+        setImageWidth(imgWidth);
+        setIsLoading(false);
+    };
+
+    const onStartShouldSetPanResponder = () => {
```

Comment:
At Expensify each handler should have a name like what it does, not what it handles.